barcodes_generator_product/views/view_product_template.xml

```diff
@@ -0,0 +1,87 @@
+Copyright (C) 2016-Today La Louve (http://www.lalouve.net)
+@author Sylvain LE GAL (https://twitter.com/legalsylvain)
+License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+-->
+<odoo>
+    <record id="view_product_template_form" model="ir.ui.view">
+        <field name="model">product.template</field>
+        <field name="inherit_id" ref="product.product_template_only_form_view" />
+        <field name="priority">100</field>
+        <field name="arch" type="xml">
+            <field name="barcode" position="before">
+                <field
+                    name="barcode_rule_id"
+                    domain="[('generate_model', '=', 'product.product')]"
+                    groups="barcodes_generator_abstract.generate_barcode"
+                    attrs="{'invisible': [('product_variant_count', '&gt;', 1)]}"
+                    colspan="2"
+                />
+                <field name="generate_type" invisible="1" />
+                <label
+                    for="barcode_base"
+                    attrs="{'invisible': ['|', ('barcode_rule_id', '=', False), ('product_variant_count', '&gt;', 1)]}"
+                    groups="barcodes_generator_abstract.generate_barcode"
+                />
+                <div
+                    name="div_barcode_base"
+                    class="o_row"
+                    attrs="{'invisible': ['|', ('barcode_rule_id', '=', False), ('product_variant_count', '&gt;', 1)]}"
+                    groups="barcodes_generator_abstract.generate_barcode"
+                >
+                    <field
+                        name="barcode_base"
+                        attrs="{'readonly': [('generate_type', '!=', 'manual')]}"
+                        groups="barcodes_generator_abstract.generate_barcode"
+                        class="oe_inline"
+                    />
+                    <button
+                        name="generate_base"
+                        type="object"
+                        string="Generate Base"
+                        help="Generate Base (Using Sequence)"
+                        attrs="{'invisible': ['|',
+                        ('generate_type', '!=', 'sequence'),
+                        ('barcode_base', '!=', 0),
+                        ]}"
+                        groups="barcodes_generator_abstract.generate_barcode"
+                        class="oe_inline"
+                    />
+                </div>
+            </field>
+
+            <field name="barcode" position="replace">
+                <label
+                    for="barcode"
+                    groups="barcodes_generator_abstract.generate_barcode"
+                    attrs="{'invisible': [('product_variant_count', '&gt;', 1)]}"
+                />
+                <div
+                    name="div_barcode"
+                    class="o_row"
+                    attrs="{'invisible': [('product_variant_count', '&gt;', 1)]}"
+                >
+                    <field
```

Comment:
Checking the behaviour of the model I found a problem. When creating a new user, the group barcodes_generator_abstract.generate_barcode is disabled. If you enter to the Variant Products form view, the label of the field will not appear but the <field> yes, making the form view look wrong for the final user.

Reply:
@legalsylvain The installation of this module dirty the view if the problem with the groups is not fixed. Could you check it?